beacon-chain/rpc/proposer_server.go

```diff
@@ -119,16 +120,20 @@ func (ps *ProposerServer) PendingAttestations(ctx context.Context, req *pb.Pendi
 	attsWithinBoundary := make([]*pbp2p.Attestation, 0, len(atts))
 	for _, att := range atts {
 
+		if err := blocks.VerifyAttestation(beaconState, att, false); err != nil {
+			log.WithField("headRoot", att.Data.Slot-
+				params.BeaconConfig().GenesisSlot).Warn("Skipping, pending attestation failed verification")
+			continue
+		}
+
 		var expectedJustifedEpoch uint64
 		if helpers.SlotToEpoch(att.Data.Slot+1) >= helpers.SlotToEpoch(currentSlot) {
 			expectedJustifedEpoch = beaconState.JustifiedEpoch
 		} else {
 			expectedJustifedEpoch = beaconState.PreviousJustifiedEpoch
 		}
 
-		if att.Data.Slot > boundary &&
-			att.Data.JustifiedEpoch == expectedJustifedEpoch &&
-			ps.chainService.IsCanonical(att.Data.Slot, att.Data.BeaconBlockRootHash32) {
+		if att.Data.Slot > boundary && att.Data.JustifiedEpoch == expectedJustifedEpoch {
```

Comment:
Do we need this if verify attestation handles all of the validity checks?

Reply:
We don't, I'll remove it. `VerifyAttestation` handles those 2 checks already

Reply:
Good point